### ISSUE TITLE
Java 11 support

### ISSUE THREAD

**Issue by nevillelyh** [spotify/scio#2174]
Placeholder to track Java 11 support. Currently blocked by https://issues.apache.org/jira/browse/BEAM-2530.

**Comment by regadas** [comment]
Working on this a bit now. I'll add more info later.

**Comment by regadas** [comment]
Here's the current scio support of `JDK11`; Note that Google Dataflow default containers are still running jdk8; for this reason and some other details in the [issue](https://issues.apache.org/jira/browse/BEAM-2530) above we still target jvm-1.8

- ✅ `scio-*` compile (scala 2.12 only)
- ✅ `scio-*` test
- ✅ `scio-repl` (sbt run & standalone)
- ✅   sbt job run
- ❌  packaged job run (still requires `jdk8`)

**Comment by jaceklaskowski** [comment]
I'm hoping this is a good place to make a comment on the main readme page. Sorry if not.

I've got Java 12 as the default JVM. While following https://github.com/spotify/scio#quick-start and running the included word count example I ran into "unsafe access" issues.

I think it'd be very helpful for newcomers like myself to add a note that scio requires Java 8 (or whatever latest version of Java is supported).

**Comment by regadas** [comment]
Hi @jaceklaskowski indeed I see the same behavior with `jdk12` around one of beam's dependencies but not with `jdk11`.

**Comment by jaceklaskowski** [comment]
How do we go about adding a note about Java 8 as a requirement?

**Comment by regadas** [comment]
@jaceklaskowski just added it. Thx!

**Comment by labianchin** [comment]
@kanterov and I were able to run Scio on JDK11 on Dataflow by doing something like this https://stackoverflow.com/a/54764118/1046584 .

Master branch on Beam SDK already has a newer way to detect `filesToStage`:

- https://github.com/apache/beam/commit/7e5a885d854eb86b7dd07f778c05fa5d595adc7d
- https://github.com/apache/beam/pull/10410
- https://github.com/apache/beam/pull/10268

I hope that would be available around Beam SDK 2.19.

**Comment by regadas** [comment]
Hi @labianchin yeah I saw the `classgraph` change as well really hope it's released soon.

IIRC `scio` used to find files to stage kinda of the same way. However we decided to move way from it and just use `PipelineResources` for a couple of reasons and decided to wait for Google/Dataflow to put everything in place:

* Dataflow harness still runs JDK8 by default; We had to set an experimental flag to actually switch to a jdk11. Not ideal for uses. In master the image  is changed [automatically](https://github.com/apache/beam/blob/9629ff6c5bfd70770046468adf44e2eeb69e606d/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java#L1922).

* A few releases back a lot of libraries were not compatible with JDK11 as well; `ByteBuddy` comes to my mind; This is not the case anymore and that's why we are able to compile/test with JDK11.

So everything coming up together nicely 😄
